### PR TITLE
Implement find contact command for ContactList

### LINKED ISSUE #240: Contact List: Implement a command to find a contact in the contact list by name

### PR DESCRIPTION
**Pull request by jonasngs**
Fixes #240 

This PR relates to the implementation of a command to find a contact in the ContactList by name, following the earlier PR for Contact and ContactList refactoring.

The following classes were added:

1. FindContactCommand: to handle the finding of a Contact from the ContactList by name.
2. FindContactParser: to handle parsing of user input for a command to find a contact by name. This class is responsible for error handling when the user provides invalid or incorrect arguments.

The following classes were modified:

1. AddressBookParser: Add a case to handle find contact command .

### CODE REVIEW

**Review by MatthiasLHK — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/FindContactParser.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindContactCommand.MESSAGE_USAGE));
+        }
+
+        String[] nameKeywords = trimmedArgs.split("\\s+");
+
+        List<String> keywords = new ArrayList<>(Arrays.asList(nameKeywords));
+        NameContainsKeywordsPredicate predicate = new NameContainsKeywordsPredicate(keywords);
```
Comment (michael-setia) [inline src/main/java/seedu/address/logic/parser/FindContactParser.java:33]: I like how this is separated into 2 lines (not combined into 1 line like the original code) which can enhance readability 😄

**Review by michael-setia — APPROVED** [review]
The implementation is similar with the rest of the code. Overall, looks great to merge 👍

File: src/main/java/seedu/address/logic/commands/FindContactCommand.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all contacts whose names contain any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " alice bob charlie";
+
+    private final NameContainsKeywordsPredicate predicate;
+
+    public FindContactCommand(NameContainsKeywordsPredicate predicate) {
+        this.predicate = predicate;
+    }
```
[the remark this answers is not in the document]
Comment (jonasngs) [reply src/main/java/seedu/address/logic/commands/FindContactCommand.java:26]: Yes, javadocs will be added.

**Pull request merged**